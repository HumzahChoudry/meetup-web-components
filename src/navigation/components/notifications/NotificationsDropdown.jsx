// @flow
import React from 'react';

import Section from '../../../layout/Section';
import Chunk from '../../../layout/Chunk';

import Notification from './Notifications';

export const NotificationsDropdownComponent = ({
	self,
	notifications,
	onMarkReadAction,
	localeCode,
	emptyContentLabel,
	notificationTitleLabel,
	generateClassicUrl,
}) => {
	const isEmptyState = notifications.length < 1;

	const NotificationsList = () => (
		<div className="notifications--populated">
			<ul className="list">
				{notifications.map((notif, key) => (
					// TODO: refactor
					// eslint-disable-next-line react/no-array-index-key
					<li key={key}>
						<Notification
							className={key === 0 ? 'border--none' : ''}
							id={notif.id}
							memberId={self.id.toString()}
							kind={notif.kind}
							photoUrl={notif.photo ? notif.photo.photo_link : ''}
							isRead={notif.read}
							link={notif.link}
							onMarkReadAction={onMarkReadAction}
							dangerouslySetInnerHTML={{ __html: notif.text }}
							updated={notif.updated}
							localeCode={localeCode}
							generateClassicUrl={generateClassicUrl}
						/>
					</li>
				))}
			</ul>
		</div>
	);

	const NotificationsEmpty = () => (
		<Section className="notifications--empty align--center valignChildren--center">
			<Chunk>
				<p>{emptyContentLabel}</p>
			</Chunk>
		</Section>
	);

	return (
		<div className="align--left">
			<h2 className="padding--all border--bottom text--labelSecondary">
				{notificationTitleLabel}
			</h2>
			{isEmptyState ? <NotificationsEmpty /> : <NotificationsList />}
		</div>
	);
};

export default NotificationsDropdownComponent;
