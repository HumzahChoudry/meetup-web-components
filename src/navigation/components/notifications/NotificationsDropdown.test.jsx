import React from 'react';
import { shallow } from 'enzyme';

import {
	MOCK_NOTIFICATION_COMMENT,
	MOCK_NOTIFICATIONS_LIST,
} from 'meetup-web-mocks/lib/notifications/api';

import { NotificationsDropdownComponent } from './NotificationsDropdown';

const notifs = [
	...MOCK_NOTIFICATIONS_LIST,
	{
		...MOCK_NOTIFICATION_COMMENT,
		photo: null,
	},
];

describe('Notifications Dropdown', () => {
	const wrapper = (props = {}) =>
		shallow(
			<NotificationsDropdownComponent
				self={{ id: '1234' }}
				notifications={notifs}
				{...props}
			/>
		);

	it('should match the snapshot with notifications ', () => {
		expect(wrapper()).toMatchSnapshot();
	});
	it('should match the snapshot with an empty notifications', () => {
		expect(wrapper({ notifications: [] })).toMatchSnapshot();
	});
});
