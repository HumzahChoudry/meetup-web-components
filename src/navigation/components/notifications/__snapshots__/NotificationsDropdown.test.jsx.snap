// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Dropdown should match the snapshot with an empty notifications 1`] = `
<div
  className="align--left"
>
  <h2
    className="padding--all border--bottom text--labelSecondary"
  />
  <NotificationsEmpty />
</div>
`;

exports[`Notifications Dropdown should match the snapshot with notifications  1`] = `
<div
  className="align--left"
>
  <h2
    className="padding--all border--bottom text--labelSecondary"
  />
  <NotificationsList />
</div>
`;
